dagger/conf.nim

```diff
@@ -47,17 +49,38 @@ type
       defaultValue: noCommand }: StartUpCommand
 
     of noCommand:
-      listenAddrs* {.
-        desc: "Specifies one or more listening multiaddrs for the node to listen on."
-        defaultValue: @[MultiAddress.init("/ip4/0.0.0.0/tcp/0").tryGet()]
-        defaultValueDesc: "/ip4/0.0.0.0/tcp/0"
-        abbr: "a"
-        name: "listen-addrs" }: seq[MultiAddress]
+      listenPorts* {.
+        desc: "Specifies one or more listening ports for the node to listen on."
+        defaultValue: @[Port(0)]
+        defaultValueDesc: "0"
+        abbr: "l"
+        name: "listen-ports" }: seq[Port]
+
+      # TODO We should have two options: the listen IP and the public IP
+      # Currently, they are tied together, so we can't be discoverable
+      # behind a NAT
+      listenIp* {.
+        desc: "The public IP"
+        defaultValue: ValidIpAddress.init("0.0.0.0")
+        defaultValueDesc: "0.0.0.0"
+        abbr: "i"
+        name: "listen-ip" }: ValidIpAddress
+
+      discoveryPort* {.
+        desc: "Specify the discovery (UDP) port"
+        defaultValue: Port(8090)
+        defaultValueDesc: "8090"
+        name: "udp-port" }: Port
+
+      netPrivKeyFile* {.
+        desc: "Source of network (secp256k1) private key file (random|<path>)"
+        defaultValue: "random"
+        name: "net-privkey" }: string
 
       bootstrapNodes* {.
         desc: "Specifies one or more bootstrap nodes to use when connecting to the network."
         abbr: "b"
-        name: "bootstrap-nodes" }: seq[MultiAddress]
+        name: "bootstrap-nodes" }: seq[string]
```

Comment:
Why string here? The serialization knows how to convert between strings and multiaddrs.

Reply:
It's a signed peer record, not a MA
We could use a MA, it wouldn't be less safe, I think

Reply:
Is it serialized? In that case we should add serialize/deserialize for those and use the proper signed record types instead.